pkg/code-generator/codegen/templates/resource_client_template.go

```diff
@@ -153,7 +156,11 @@ func (client *{{ lower_camel .Name }}Client) Watch(namespace string, opts client
 			select {
 			case resourceList := <-resourcesChan:
 				select {
-					case {{ lower_camel .PluralName }}Chan <- convertTo{{ .Name }}(resourceList):
+{{- if eq .Name "FakeResource" }}
```

Comment:
I like how you guard this test case so we only generate this for the specific resource that occurs in our test. However, I think it's not ideal that we need to have our template have logic that is specific to a test case.

In the below branch, I outline an alternate proposal, for creating a new client that wraps the existing FakeResourceClient, and injects latency into that client:
https://github.com/solo-io/solo-kit/compare/emitter-map-follow-up...sam/emitter-map-follow-up?expand=1

Let me know what you think!

Reply:
pretty thrilled with this suggestion--I (before) didn't have the knowledge on how to make this in-place in the testing file.

The implementation you threw up is quite nice!  I'll steal it

Reply:
oh yeah!
* great point allowing it to be customizable per namespace
* I'm thrilled we no longer need resource-specific templating